### ISSUE TITLE
ReferenceError: HTMLElement is not defined

### ISSUE THREAD

**Issue by wildlifechorus** [tjoskar/ng-lazyload-image#271]
Suddenly I start getting this error, both with 3.3.3 and 3.3.5/6.

/var/www/html/node_modules/ng-lazyload-image/src/lazyload-image.directive.js:76
        __metadata("design:type", HTMLElement)
                                  ^

ReferenceError: HTMLElement is not defined
    at /var/www/html/node_modules/ng-lazyload-image/src/lazyload-image.directive.js:76:35
    at Object.<anonymous> (/var/www/html/node_modules/ng-lazyload-image/src/lazyload-image.directive.js:101:2)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/var/www/html/dist/stag/server/main.bundle.js:2236:18)

ng -v returns the following:

angular/cli: 1.4.3

node: 7.7.1

os: linux x64

angular/animations: 4.4.6

angular/common: 4.4.6

angular/compiler: 4.4.6

angular/core: 4.4.6

angular/forms: 4.4.6

angular/http: 4.4.6

angular/platform-browser: 4.4.6

angular/platform-browser-dynamic: 4.4.6

angular/platform-server: 4.4.6

angular/router: 4.4.6

angular/cli: 1.4.3

angular/compiler-cli: 4.4.6

angular/language-service: 4.4.6

typescript: 2.2.2

**Comment by tjoskar** [comment]
Hi @wildlifechorus, 
How does your tsconfig.json look like?

**Comment by wildlifechorus** [comment]
tsconfig.json
```
{
  "compilerOptions": {
    "baseUrl": "",
    "declaration": false,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "lib": ["es7", "dom"],
    "mapRoot": "./",
    "module": "es6",
    "moduleResolution": "node",
    "outDir": "../dist/prod",
    "sourceMap": true,
    "target": "es5",
    "typeRoots": [
      "../node_modules/@types"
    ]
  }
}

```

tsconfig.server.json

```
{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "outDir": "../dist/prod/server",
    "module": "commonjs"
  },
  "exclude": [
    "test.ts",
    "**/*.spec.ts"
  ],
  "angularCompilerOptions": {
    "entryModule": "app/app-server.module#AppServerModule"
  }
}
```

**Comment by wildlifechorus** [comment]
This error is only triggered on production/staging.

**Comment by tjoskar** [comment]
@wildlifechorus, this seams to be a bug. When you are server rendering the app `HTMLElement` is not defined (in node). I can't say why or when `HTMLElement` was included.

**Comment by wildlifechorus** [comment]
But is it related to ng-lazyload-image or has something to do with my app? Because the error states that it's coming from ng-lazyload-image/src/lazyload-image.directive.js:76

**Comment by tjoskar** [comment]
I believe it's a bug in `ng-lazyload-image` and that it was introduced in this commit: https://github.com/tjoskar/ng-lazyload-image/commit/26fc5f03a80ce5fcdd4cd81f010288ed43def169#diff-6357700a89283deb3dc4ed8b892af91aR40

But it should in that case work with version 3.4.0 and below but you are already using `3.3.3`? Can you run `npm ls ng-lazyload-image` just to be sure?

**Comment by wildlifechorus** [comment]
Returns the following: `ng-lazyload-image@3.3.3`

When I remove the module all together, everything works properly.

BTW already removed `dist` and `node_modules` and made npm install again...no luck.

**Comment by tjoskar** [comment]
Hum, that is weird. In the latest version (3.4.1) the following is on line 76 in lazyload-image.directive.js: `__metadata("design:type", HTMLElement)` which is a bug since `HTMLElement` is not defind in node.js. However in version 3.3.3 the following line `__metadata("design:type", Object)` is present which should be fine. 

Can you check your `/var/www/html/node_modules/ng-lazyload-image/src/lazyload-image.directive.js` at line 76 to see what you got? It does not make any sense to me.

And just to be clear, you are getting this error on your local machine, the same machine that you run `npm ls ng-lazyload-image` on?

**Comment by wildlifechorus** [comment]
Looks like I'm using "ng-lazyload-image@3.4.1" on the staging and production machine, that's maybe why it breaks.

So in `package.json` maybe I should change `"ng-lazyload-image": "^3.3.3",` to `"ng-lazyload-image": "3.3.3",` No?

**Comment by wildlifechorus** [comment]
Fixed my problem :D

**Comment by tjoskar** [comment]
> So in package.json maybe I should change "ng-lazyload-image": "^3.3.3", to "ng-lazyload-image": "3.3.3", No?

Yeah, or even better: use npm >= 5 with [package-lock.json](https://docs.npmjs.com/files/package-lock.json) or [yarn](https://yarnpkg.com/en/) :)

> Fixed my problem

Great! I will however keep this issue open since it is a bug in version 3.4.1 with server rendering.

**Comment by sapierens** [comment]
I'll take a look at this tomorrow.

**Comment by tjoskar** [comment]
@sapierens, it may be a better way but changing the type of: https://github.com/tjoskar/ng-lazyload-image/blob/eda0c82f82b4371b6e6fcb000f662be09532f21c/src/lazyload-image.directive.ts#L40

to `any` or `object` is good enough for me.

**Comment by sapierens** [comment]
> to `any` or `object` is good enough for me.

That's what I ended up doing - #273.

**Comment by tjoskar** [comment]
A fix is included in 3.4.2, thanks @sapierens
